### PR TITLE
Feature/javascript improvements1

### PR DESCRIPTION
**Pull request by MikeVanMourik**
asana: https://app.asana.com/0/1201027711166952/1205853947344178/f

Some basic changes to the javascripts,

**PLEASE "be strict" im trying to improve my javascript knowledge**

these changes are just const and let changes in a few files, I looked at the functions and calls to see if i could improve those but none seemed like they would need refactoring. (if i missed something please let me know )

i used const in quite a few places and tried this in wiserdemo i did not notice any issues, however. most scripts have a {costumscript} parameter at the bottom, should i instead use let so that items are mutable by this?

### CODE REVIEW

File: Api/Modules/Items/FieldTemplates/date-time picker.js
Comment (GilianJuice) [inline Api/Modules/Items/FieldTemplates/date-time picker.js:1]: You forgot a `var` at line 12.

**Review by GilianJuice — CHANGES_REQUESTED** [review]
The changes you made look good, but you should do some more changes:
* Please use arrow functions, that is the new standard with ES6. So for example, instead of `function(event) {`, you should use `(event) => {`.
* Use template strings instead of concattenation. So not `dynamicItems.settings.wiserApiRoot + "items/" + encodeURIComponent("{itemIdEncrypted}") + "/action-button/{propertyId}?queryId=" + encodeURIComponent(optionsFromProperty.queryId || 0)`, but `${dynamicItems.settings.wiserApiRoot}items/${encodeURIComponent("{itemIdEncrypted}")}/action-button/{propertyId}?queryId=${encodeURIComponent(optionsFromProperty.queryId || 0)}`. (To use template strings, you need to use backticks instead of quotes.)
* Use await/async instead of `.then()` etc.

**Comment by GilianJuice** [comment]
You don't have to use `let`, most variables should not be changed by custom scripts anyway. And people can always copy it to a new variable if they really need to change it.

**Comment by MikeVanMourik** [comment]
@GilianJuice I made the new changes, could you give me some pointers on the await catch try logic.
I'm not 100% if this still the same. i tried a quick file upload on wiserdemo and it seemed to work but I think .then logic doesnt fire when there is a error but "finally" does so there might be issues there?
also the error is just a single paramater and not the 3 we used to have, i just removed them because i think the api function already logs those values anyway

**Comment by MikeVanMourik** [comment]
Alright, i am aware of the "this" scope i already ran into a issue with with a bugfix once and then hans helped me out

i moved the logic away from the finaly block and into the try as per your request.

i think this set of javascripts is done unless you have further tips suggestions, so ill start on javascriptimprovements2

**Comment by MikeVanMourik** [comment]
@GilianJuice any further improvements needed for these files? :)

**Comment by GilianJuice** [comment]
@MikeVanMourik Yes, you forgot to change one function to an arrow function. Please fix that, then we can merge this PR.

**Comment by MikeVanMourik** [comment]
ah sneaky little bastard, found it and replaced

**Review by GilianJuice — APPROVED** [review]

**Pull request merged**